XHR-lib.pl

```diff
@@ -0,0 +1,48 @@
+#!/usr/local/bin/perl
+# A pile of XHR related routines
+
+use strict;
+
+our (%in, %gconfig, $root_directory, $remote_user, $current_theme);
+
+sub xhr
+{
+
+my %data    = ();
+my $output  = sub {
+    my ($data) = @_;
+    print "x-no-links: 1\n";
+    print_json($data);
+    };
+
+# Fetch actions
+if ($in{'action'} eq "fetch") {
```

Comment:
What happens if none of these if statements match? Shouldn't we call `error` ?

Reply:
> What happens if none of these if statements match? Shouldn't we call `error` ?

Probably, yes! It would be a better design. I'll add error response in JSON..